src/mixpanel-core.js

```diff
@@ -696,10 +698,16 @@ MixpanelPersistence.prototype._get_or_create_queue = function(queue, default_val
     return this['props'][key] || (this['props'][key] = default_val);
 };
 
-MixpanelPersistence.prototype.set_event_timer = function(event_name, timestamp) {
+MixpanelPersistence.prototype.set_event_timer = function(event_name, timestamp, callback) {
     var timers = this['props'][EVENT_TIMERS_KEY] || {};
     timers[event_name] = timestamp;
     this['props'][EVENT_TIMERS_KEY] = timers;
+
+    if (callback && (typeof(callback) === 'function')) {
+        var callbacks = this['props'][EVENT_TIMERS_KEY_CALLBACK] || {};
+        callbacks[event_name] = callback;
+        this['props'][EVENT_TIMERS_KEY_CALLBACK] = callbacks;
```

Comment:
Unfortunately, persisted props need to be serializable - they end up either in a cookie or localStorage (depending on your configuration). The reason the event-timer code currently uses the persistent store is so that it can still work across page loads, but I don't think we'll be able to do that with an arbitrary callback, at least without some wonky workarounds like registering the callback code as part of `mixpanel.init()` configuration.

If, however, you don't care about the cross-pageload case, then we can avoid all the `EVENT_TIMERS_KEY_CALLBACK` business and skip the persistent store altogether, just attach the callback to the lib instance directly.

Reply:
That makes sense, thanks for the review. I don't think it's worth supporting the cross-pageload case, the hacky implementation nonsense relative to benefit offered (I certainly wouldn't be using it) just doesn't justify it right now. I think I see how instances work, so I'll take a stab at this now, doesn't look like it'll take long.